blackhole.c

```diff
@@ -717,17 +712,17 @@ void blackhole_make_one(int index) {
 
     int child = domain_fork_particle(index);
 
-    NLocal[5] ++;
     P[child].PI = atomic_fetch_and_add(&N_bh_slots, 1);
     P[child].Type = 5;	/* make it a black hole particle */
 
-    P[child].StarFormationTime = All.Time;
+    BHP(child).FormationTime = All.Time;
     /*Ensure that mass is conserved*/
     double BHmass = All.SeedBlackHoleMass;
     if(P[index].Mass <= All.SeedBlackHoleMass) {
         P[index].IsGarbage = 1;
         BHmass = P[index].Mass;
     }
+    BHP(child).Generation = SPHP(index).Generation;
```

Comment:
I still feel Generation shall be a common variable in P. It is part of book keeping in forking particles -- and forking should be OK on any particles.